sw/device/tests/kmac_idle_test.c

```diff
@@ -0,0 +1,127 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_clkmgr.h"
+#include "sw/device/lib/dif/dif_kmac.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_clkmgr_t clkmgr;
+static dif_kmac_t kmac;
+
+const test_config_t kTestConfig;
+
+const dif_clkmgr_hintable_clock_t kmac_clock =
+    kTopEarlgreyHintableClocksMainKmac;
+
+// Digest lengths in 32-bit words.
+#define DIGEST_LEN_SHA3_256 (256 / 32)
+#define DIGEST_LEN_SHA3_512 (512 / 32)
+#define DIGEST_LEN_SHA3_MAX DIGEST_LEN_SHA3_512
+
+// SHA-3 test description.
+typedef struct sha3_test {
+  dif_kmac_mode_sha3_t mode;
+
+  const char *message;
+  size_t message_len;
+
+  const uint32_t digest[DIGEST_LEN_SHA3_MAX];
+  size_t digest_len;
+} sha3_test_t;
+
+// SHA-3 test.
+const sha3_test_t sha3_256_test = {
+    // Example taken from NIST FIPS-202 Algorithm Test Vectors:
+    // https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Algorithm-Validation-Program/documents/sha3/sha-3bytetestvectors.zip
+    .mode = kDifKmacModeSha3Len256,
+    .message = "\xe7\x37\x21\x05",
+    .message_len = 4,
+    .digest = {0x8ab6423a, 0x8cf279b0, 0x52c7a34c, 0x90276f29, 0x78fec406,
+               0xd979ebb1, 0x057f7789, 0xae46401e},
+    .digest_len = DIGEST_LEN_SHA3_256};
+
+static void set_hint(dif_toggle_t clock_hint_state) {
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, kmac_clock,
+                                                  clock_hint_state));
+}
```

Comment:
Nit: this static helper function feels unnecessary. You could probably just use
```c
CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, kmac_clock, ...));
```
inline throughout below.

Reply:
Thanks Tim, I was in two minds whether to leave the function or put the code inline. I have made the change.

Reply:
No problem, thanks!